graph.cpp

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+#include <list>
+#include <map>
+using namespace std;
+
+
+class Node{
+    public:
+        int nodeID;
+        // Placeholder for 3D location info as I'm not sure how Godot handles it
+        tuple <double, double, double> location;
+        // Placeholder for formula data
+        int data;
+        list<Node*> neighbors;
+
+        Node(){
+            nodeID = 0;
+            location = make_tuple(0,0,0);
+            data = 0; 
+        }
+        Node(int nodeID, tuple<double, double, double> location, int data){
```

Comment:
Objects (the location tuple in this case) in C++ should be passed by constant reference rather than copy where possible. You're making a copy already anyway when you assign it to the member, no need to make 2. 

```suggestion
        Node(int nodeID, const tuple<double, double, double>& location, int data){
```